wgpu/src/backend/web.rs

```diff
@@ -1113,13 +1113,21 @@ impl crate::context::Context for Context {
         _adapter: &Self::AdapterId,
         _adapter_data: &Self::AdapterData,
     ) -> wgt::SurfaceCapabilities {
-        wgt::SurfaceCapabilities {
-            // https://gpuweb.github.io/gpuweb/#supported-context-formats
-            formats: vec![
+        let format = self.0.get_preferred_canvas_format();
+        let (preferred, second) = if map_texture_format(wgt::TextureFormat::Bgra8Unorm) == format {
+            (
                 wgt::TextureFormat::Bgra8Unorm,
                 wgt::TextureFormat::Rgba8Unorm,
-                wgt::TextureFormat::Rgba16Float,
-            ],
+            )
+        } else {
+            (
+                wgt::TextureFormat::Rgba8Unorm,
+                wgt::TextureFormat::Bgra8Unorm,
+            )
+        };
```

Comment:
This code won't respect get_preferred_canvas_format if it's `Rgba16Float` or something else entirely. We should create the vector with the preferred canvas format, then insert-if-not-present the other 3 formats.